module_6/hw2.py

```diff
@@ -0,0 +1,27 @@
+# 2. Implement a map-like function that returns an iterator (generator function)
+# extra functionality: if arg function can't be applied, return element as is + text exception
+
+
+def map_like(fun, *elems):
+    for elem in elems:
+        try:
+            yield fun(elem)
+        except TypeError:
+            yield elem
```

Comment:
if exception appears, need to print also exception error 
from the task: `# 6: 'int' object is not subscriptable` that text coming from exception error message

Reply:
+

Reply:
Thank you for the comment! Fixed.